apps/studio/src/server/modules/collection/__tests__/collection.router.test.ts

```diff
@@ -0,0 +1,910 @@
+      )
+    })
+
+    it.skip("should throw 403 if user does not have write access to the parent collection", async () => {})
+  })
+  describe("getMetadata", () => {
+    it("should throw 401 if not logged in", async () => {
+      // Act
+      const result = unauthedCaller.getMetadata({
+        siteId: 1,
+        resourceId: -1,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({ code: "UNAUTHORIZED" }),
+      )
+    })
+
+    it("should throw 404 if `siteId` does not exist", async () => {
+      // Arrange
+      const invalidSiteId = 999
+      const { site } = await setupSite()
+      await setupAdminPermissions({
+        userId: session.userId,
+        siteId: site.id,
+      })
+      expect(site.id).not.toEqual(invalidSiteId)
+
+      // Act
+      const result = caller.getMetadata({
+        siteId: invalidSiteId,
+        resourceId: 1,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "NOT_FOUND",
+          message:
+            "You do not have sufficient permissions to perform this action",
+        }),
+      )
+    })
+
+    it("should throw 404 if `collectionId` does not exist", async () => {
+      // Arrange
+      const { site } = await setupSite()
+      await setupAdminPermissions({
+        userId: session.userId,
+        siteId: site.id,
+      })
+
+      // Act
+      const result = caller.getMetadata({
+        siteId: site.id,
+        resourceId: 999,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "NOT_FOUND",
+          message: "Collection not found",
+        }),
+      )
+    })
+
+    it("should throw 403 if user does not have read access to the site", async () => {
+      // Arrange
+      const { collection, site } = await setupCollection()
+
+      // Act
+      const result = caller.getMetadata({
+        siteId: site.id,
+        resourceId: Number(collection.id),
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "NOT_FOUND",
+          message:
+            "You do not have sufficient permissions to perform this action",
+        }),
+      )
+    })
+
+    it("should return 200 ", async () => {
+      // Arrange
+      const { collection, site } = await setupCollection()
+      await setupAdminPermissions({ userId: session.userId, siteId: site.id })
+
+      // Act
+      const result = await caller.getMetadata({
+        siteId: site.id,
+        resourceId: Number(collection.id),
+      })
+
+      // Assert
+      const expected = await db
+        .selectFrom("Resource")
+        .select(["Resource.title", "Resource.permalink", "Resource.parentId"])
+        .where("id", "=", collection.id)
+        .executeTakeFirst()
+      expect(result).toMatchObject(expected!)
+    })
+  })
+
+  describe("updateCollectionLink", () => {
```

Comment:
nit - can standardize with the other test suites to
- add line breaks between the tests
- add labelling for Arrange,Act,Assert
- add another test for non-logged in users
- name the test "should throw XXX error..." instead of "should fail..."